parse.go

```diff
@@ -0,0 +1,76 @@
+package github
+
+import (
+	"errors"
+	"flag"
+	"io"
+	"os"
+	"strconv"
+)
+
+var (
+	appId         int64
+	instId        int64
+	path          string
+	noKeyError    = "you need to pass the private key either with '-k' parameter or by setting 'GITHUB_KEY_PATH' or 'GITHUB_KEY_VALUE' or by passing it via STDIN\n"
+	noAppIdError  = "You need to define the App ID via '-a' parameter or 'GITHUB_APP_ID' environment variable\n"
+	noInstIdError = "You need to define the Installation ID via '-i' parameter or 'GITHUB_INST_ID' environment variable\n"
+)
+
+func ParseParameters() (*GitHubAppConfig, error) {
+	var err error
+	cfg := &GitHubAppConfig{}
+
+	flag.StringVar(&path, "k", "", "Path to key file for authentication")
+	flag.Int64Var(&appId, "a", 0, "App ID to use for authentication")
+	flag.Int64Var(&instId, "i", 0, "Installation ID that identifies the APP installation ID on GitHub")
+	flag.Parse()
+
+	// Get the values from the environment variables if they are set with parameters
+	if appId == 0 {
+		appId, err = strconv.ParseInt(os.Getenv("GITHUB_APP_ID"), 10, 64)
+		if err != nil || appId == 0 {
+			return nil, errors.New(noAppIdError)
+		}
+	}
+	if instId == 0 {
+		instId, err = strconv.ParseInt(os.Getenv("GITHUB_INST_ID"), 10, 64)
+		if err != nil || instId == 0 {
+			return nil, errors.New(noInstIdError)
+		}
+	}
+
+	// Get the key from environment variables
+	cfg.PrivateKey = []byte(os.Getenv("GITHUB_KEY_VALUE"))
```

Comment:
How does osGetenv cope with an EnvVar that does not exist?  Is an error returned?  Does it treat it as an empty string?